test/txstart-test.js

```diff
@@ -0,0 +1,225 @@
+    });
+
+    const block = new Block();
+    block.txs.push(tx1);
+    block.txs.push(tx2);
+    block.prevBlock = node.chain.tip.hash;
+    block.time = node.chain.tip.time + 2;
+    block.bits = await node.chain.getTarget(block.time, node.chain.tip);
+
+    await assert.rejects(node.chain.add(block, VERIFY_NONE),
+      {reason: 'no-tx-allowed-yet'});
+  });
+
+  it('Should accept empty block before txStart', async () => {
+    // Create an address that takes literally nothing to spend
+    const addr = Address.fromScript(new Script());
+
+    const tx1 = new TX({
+      inputs: [new Input()],
+      outputs: [new Output()]
+    });
+    tx1.outputs[0].address = addr;
+    tx1.locktime = node.chain.height + 1;
+
+    const block = new Block();
+    block.txs.push(tx1);
+    block.prevBlock = node.chain.tip.hash;
+    block.time = node.chain.tip.time + 3;
+    block.bits = await node.chain.getTarget(block.time, node.chain.tip);
+
+    await node.chain.add(block, VERIFY_NONE);
+
+    // Spend this output later
+    utxo = block.txs[0].hash();
+  });
+
+  it('Should add blocks until txStart is reached', async() => {
+    for (let i = node.chain.height; i < node.network.txStart - 1; i++) {
+      const block = await node.miner.mineBlock();
+      assert(await node.chain.add(block));
+    }
+    // The NEXT block is the txStart height
+    assert.strictEqual(node.chain.height + 1, node.network.txStart);
+  });
+
+  it('Should allow TX in mempool at txStart height', async () => {
+    lastTX = new TX({
+      inputs: [new Input()],
+      outputs: [new Output()]
+    });
+    lastTX.inputs[0].prevout.hash = utxo;
+    lastTX.inputs[0].prevout.index = 0;
+    lastTX.inputs[0].witness.items[0] = Buffer.from([1, 1]);  // true
+    lastTX.inputs[0].witness.items[1] = Buffer.alloc(0);      // empty script
+
+    await node.mempool.addTX(lastTX);
+    assert.strictEqual(node.mempool.map.size, 1);
+    assert(node.mempool.has(lastTX.hash()));
+  });
+
+  it('Should allow Claim in mempool at txStart height', async () => {
+    const claim = await wallet.fakeClaim('cloudflare');
+
+    try {
+      ownership.ignore = true;
+      await node.mempool.addClaim(claim);
+    } finally {
+      ownership.ignore = false;
+    }
+    assert.strictEqual(node.mempool.claims.size, 1);
+    assert(node.mempool.hasClaim(claim.hash()));
+  });
+
+  it('Should allow Airdrop in mempool at txStart height', async () => {
+    await node.mempool.addAirdrop(proof);
+    assert.strictEqual(node.mempool.airdrops.size, 1);
+    assert(node.mempool.hasAirdrop(proof.hash()));
+  });
+
+  it('Should accept a block full of goodies at txStart height', async() => {
+    const block = await node.miner.mineBlock();
+    try {
+      ownership.ignore = true;
+      assert(await node.chain.add(block));
+    } finally {
+      ownership.ignore = false;
+    }
+
+    // This is the first block where transactions are allowed
+    assert.strictEqual(node.chain.height, node.network.txStart);
+
```

Comment:
A comment summarizing the contents of the block would be helpful here, in particular the outputs in the coinbase.